platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -292,15 +293,25 @@ private void captureTimestamp(final ConsensusEvent event) {
      * @param transaction the transaction to apply
      */
     private void handleTransaction(final ConsensusTransaction transaction) {
-        if (transaction.isSystem()) {
+        if (isSystemTransaction(transaction.getApplicationTransaction())) {
             return;
         }
+
         final int delta =
                 ByteUtils.byteArrayToInt(transaction.getApplicationTransaction().toByteArray(), 0);
         runningSum += delta;
         setChild(RUNNING_SUM_INDEX, new StringLeaf(Long.toString(runningSum)));
     }
 
+    private boolean isSystemTransaction(final Bytes transactionBytes) {
+        if (transactionBytes == null || transactionBytes.length() == 0) {
+            return false;
+        }
+
+        final var transaction = transactionBytes.toByteArray();
```

Comment:
`toByteArray` makes a copy which is not needed for checking its length, `transactionBytes.length()` is enough

Reply:
Fixed